gimme_aws_creds/config.py

```diff
@@ -197,16 +197,57 @@ def get_config_dict(self, include_inherits = True):
             config = configparser.ConfigParser()
             config.read(self.OKTA_CONFIG)
 
-            try:
-                profile_config = dict(config[self.conf_profile])
-                self.fail_if_profile_not_found(profile_config, self.conf_profile, config.default_section)
-                return self._handle_config(config, profile_config, include_inherits)
-            except KeyError:
-                if self.action_configure:
-                    return {}
-                raise errors.GimmeAWSCredsError(
-                    'Configuration profile not found! Use the --action-configure flag to generate the profile.')
-        raise errors.GimmeAWSCredsError('Configuration file not found! Use the --action-configure flag to generate file.')
+            selected_section = None
+            if self.conf_profile is not None:
+                try:
+                    selected_section = config[self.conf_profile]
+                    if self.conf_profile == 'DEFAULT':
+                        found_default_option = False
+                        for _ in selected_section:
+                            found_default_option = True
+                            break
+                        if not found_default_option:
+                            selected_section = None
+
+                except KeyError:
+                    raise errors.GimmeAWSCredsError(
+                        'Configuration profile not found! Did you spell the profile name incorrectly? If not, '
+                        'use the --action-configure flag to generate a profile named {}.'.format(self.conf_profile))
+            else:
+                self.ui.message('The --profile switch has not been set. Searching for an automatic profile...')
+
+            if selected_section is None:
+                if self.ui.environ.get('AWS_PROFILE') is not None:
+                    aws_env_profile = self.ui.environ.get('AWS_PROFILE')
+                    self.ui.message('The AWS_PROFILE environment variable is set. Searching for a profile '
+                                    'where `aws_profile_env_match = {}`'.format(aws_env_profile))
+                    for section_name in config.sections():
+                        section = config[section_name]
+                        # Use a different key than `cred_profile` for e.g., cases where gimme-aws-creds
+                        # manages base credentials but elevated credentials are acquired via the STS AssumeRole
+                        # action using a different AWS profile name
+                        if 'aws_profile_env_match' in section and section['aws_profile_env_match'] == aws_env_profile:
```

Comment:
I don't believe this is working for me in tests.
config.sections() contains an array of names like ['test1', 'test1']
When we get here, "section" is a configparser <Section> item but it does not contain anything like `aws_profile_env_match` as an item.
Makes more sense to me to just use the section name..

`for section_name in config.sections():
    if section_name == "aws_env_profile":
        ....
`
If all you are trying to do is find a section name that matches the ENV, then you are done with that check.

If you need to do an elevated check, you are going to have to probably do another check here, because this format breaks for users like me who might have a simple environment with a profile like "test1" which means that if I set AWS_PROFILE to "test1", I get no matches here because there is no 'aws_profile_env_match' in section so it will never match anything.